core/vm/contracts_lightclient.go

```diff
@@ -397,3 +401,72 @@ type cometBFTLightBlockValidateHertz struct {
 func (c *cometBFTLightBlockValidateHertz) Run(input []byte) (result []byte, err error) {
 	return c.run(input, true)
 }
+
+// tmSignatureRecover implemented as a native contract.
+type tmSignatureRecover struct{}
+
+func (c *tmSignatureRecover) RequiredGas(input []byte) uint64 {
+	return params.EcrecoverGas
+}
+
+const (
+	tmPubKeyLength       uint8 = 33
+	tmSignatureLength    uint8 = 64
+	tmSignatureMsgLength uint8 = 32
+)
+
+func (c *tmSignatureRecover) Run(input []byte) (result []byte, err error) {
```

Comment:
Please add comment in this file to explain the stucture of the inputs.

Reply:
done